### PR TITLE
Only fire exit buffering sentinel if time has changed, or paused when…

### PR DESCRIPTION
**Pull request by andrewjackson001**
… metadata loaded.

If paused and metadata not loaded then do not fire)

### CODE REVIEW

**Comment by dhurrell** [comment]
I imagine that this._hasMetaDataBeenLoaded needs to be reset whenever the player is reset, otherwise the extra check will only affect the first piece of media loaded. I'm not even convinced you need a new  member variable - _readyToPlayFrom seems to serve the same purpose.

Personally I prefer
```
function fireExitBufferingSentinel(self) ...
```

over
```
var fireExitBufferingSentinel = function(self) ...
```

But I don't think we have any coding guidance around that.

File: static/script-tests/tests/devices/mediaplayer/html5commontests.js [hunk 1]
```diff
@@ -1287,6 +1287,20 @@ window.commonTests.mediaPlayer.html5.mixinTests = function (testCase, mediaPlaye
         });
     };
 
+    mixins.testExitBufferingSentinelIsFiresWhenDeviceIsPausedAndMetadataHasBeenNotBeenLoaded = function(queue) {
```
Comment (tsadlerBBC) [inline static/script-tests/tests/devices/mediaplayer/html5commontests.js]: Needs renaming to testExitBufferingSentinelIs**Not**Fired...

**Comment by tsadlerBBC** [comment]
I agree with David that _readyToPlayFrom should be used.

File: static/script-tests/tests/devices/mediaplayer/html5commontests.js (hunk 1 above)
Comment (sdclibbery) [inline static/script-tests/tests/devices/mediaplayer/html5commontests.js]: Is there a 'not' missing in the name of this test..? :-)

Comment (rthompson1991) [inline static/script-tests/tests/devices/mediaplayer/html5commontests.js]: I think you're missing a not in this test name

**Pull request merged**